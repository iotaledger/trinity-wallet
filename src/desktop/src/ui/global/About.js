/*global Electron*/
import React from 'react';

import Modal from 'ui/components/modal/Modal';
import Button from 'ui/components/Button';
import Scrollbar from 'ui/components/Scrollbar';
import Icon from 'ui/components/Icon';

import { DESKTOP_VERSION } from 'config';

import css from './about.scss';

/**
 * About window component
 */
class About extends React.PureComponent {
    state = {
        visible: false,
    };

    componentDidMount() {
        this.onMenuToggle = this.menuToggle.bind(this);
        Electron.onEvent('menu', this.onMenuToggle);
    }

    componentWillUnmount() {
        Electron.removeEvent('menu', this.onMenuToggle);
    }

    menuToggle(item) {
        switch (item) {
            case 'about':
                this.setState({
                    visible: true,
                });
                break;
        }
    }

    render() {
        const { visible } = this.state;

        return (
            <Modal variant="global" isOpen={visible} onClose={() => this.setState({ visible: false })}>
                <section className={css.about}>
                    <Icon icon="iota" size={48} />
                    <h1>Trinity Wallet</h1>
                    <h2>
                        v{DESKTOP_VERSION} <small>BETA</small>
                    </h2>

                    <article>
                        <Scrollbar>
                            <h5>0.1.7.</h5>
                            <ul>
                                <li>New: Wallet Expert mode settings</li>
                                <li>New: Manual promote and rebroadcast functionality</li>
                                <li>New: Auto-promotion enable/disable settings</li>
                                <li>Fixed: On send incorrect progress steps displayed</li>
                                <li>Fixed: Cannot enable Two-Factor authentication</li>
                                <li>Fixed: Node error on Adding additional Seed hangs wallet</li>
                                <li>Fixed: QR seed scanning hangs wallet</li>
                            </ul>
                            <h5>0.1.6.</h5>
                            <ul>
                                <li>New: Search and Hide zero balance functionality to history view</li>
                                <li>Updated: Clear memory from plain text seed on onboarding</li>
                                <li>Updated: Restore wallet size and position on reopening</li>
                                <li>Fixed: Narrow sidebar layout on Setting and Dashboard views</li>
                                <li>Fixed: Wrong address displayed if closing Receive while generating</li>
                                <li>Fixed: Dynamic node list does not load correctly</li>
                                <li>Fixed: Adding custom node results in error</li>
                                <li>Fixed: Reverse address list order on Address list view</li>
                                <li>Fixed: on Windows wallet does not revert to original size after maximize</li>
                            </ul>
                            <h5>0.1.5.</h5>
                            <ul>
                                <li>New: Major user interface and theming update</li>
                                <li>New: Snapshot transition functionality</li>
                                <li>New: Password strength and common password check</li>
                                <li>Updated: Windows platform frameless application style</li>
                                <li>Updated: Support `iota:` and `iota://` deep links</li>
                                <li>Fixed: Amount input behaviour should match mobile</li>
                                <li>Fixed: Address list overlaps in Account settings</li>
                            </ul>

                            <h5>0.1.4.</h5>
                            <ul>
                                <li>Added password strength estimation</li>
                                <li>Windows platform specific UI update</li>
                                <li>Added step-by-step manual seed write down view</li>
                                <li>Added snapshot transition functionality</li>
                                <li>Theme color update and usage fixes</li>
                            </ul>

                            <h5>0.1.3.</h5>
                            <ul>
                                <li>New: Multi-account background polling</li>
                                <li>New: Automatic random node switch mode</li>
                                <li>New: Linux missing libraries check and setup tutorial</li>
                                <li>Updated: Windows and Linux application icon assets</li>
                                <li>Updated: Settings views unified flow and layout</li>
                                <li>Fixed: Missing character limit and validation on message field</li>
                                <li>Fixed: Adding additional seed fail crashes wallet</li>
                                <li>Fixed: On linux/windows VM can`t send transactions</li>
                                <li>Fixed: Onboarding back link crashes the wallet</li>
                                <li>Fixed: UI scrollbars ugly or missing</li>
                                <li>Fixed: Cannot copy address in Address list</li>
                                <li>Fixed: Multiple smaller UI and localisation fixes</li>
                            </ul>
                            <h5>0.1.2.</h5>
                            <ul>
                                <li>Added: background polling</li>
                                <li>Added: a bunch of missing translations</li>
                                <li>Added: auto focus on input fields</li>
                                <li>Added: option to change wallet lock time-out in Advanced settings</li>
                                <li>
                                    Updated: encrypted seed storage uses keychain on mac, Credential Vault on Windows
                                    and libsceret on linux
                                </li>
                                <li>Updated: history card ui and functionallity </li>
                                <li>Updated: chart converts market cap and volume to wallets currency</li>
                                <li>Fixed: Seed validation broken on onboarding</li>
                                <li>Fixed: Two-factor disable flow broken</li>
                                <li>Fixed: Clipboard clears on onboarding even seed is not generated</li>
                                <li>Fixed: Node settings hang forever when invalid node address provided</li>
                                <li>Fixed: History filter views show incorrect items</li>
                                <li>Fixed: Empty account name brakes wallet</li>
                                <li>Fixed: Value field conversion broken</li>
                                <li>Fixed: dark theme elements unreadable</li>
                            </ul>
                            <h5>0.1.1.</h5>
                            <ul>
                                <li>Amount value input now has a unit dropdown instead of switch</li>
                                <li>History card transaction details are contained in the card itself</li>
                                <li>Fixed: Sending transactions with value throws generic error</li>
                                <li>Fixed: Un-closable wallet on linux and windows wallet</li>
                                <li>Fixed: Two-factor authentication locks out of the wallet</li>
                                <li>Fixed: Receive address does not reset when reopening Receive view</li>
                                <li>Fixed: QR code not scannable with dark theme active</li>
                                <li>Fixed: Opaque wallet lock window background</li>
                                <li>Fixed: Amount value input incorrect fiat conversion</li>
                                <li>Fixed: Balance fiat value incorrectly rounded</li>
                                <li>Fixed: History transaction details don`t match selected item</li>
                                <li>Fixed: Seed generator view resets seed when navigating back from Backup seed</li>
                                <li>Missing translations added</li>
                                <li>Minor visual bugfixes here and there</li>
                                <li>Removed windows max size limit</li>
                            </ul>
                        </Scrollbar>
                    </article>
                    <p>
                        <strong>BETA version warning:</strong> Do not send large amounts with this wallet.
                    </p>
                    <footer>
                        <Button onClick={() => this.setState({ visible: false })} variant="dark">
                            Close
                        </Button>
                    </footer>
                </section>
            </Modal>
        );
    }
}

export default About;
