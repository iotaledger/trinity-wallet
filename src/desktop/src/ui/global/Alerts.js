import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import withAlertsData from 'containers/global/Alerts';

import Icon from 'ui/components/Icon';

import css from './alerts.scss';

/**
 * Alerts UI helper component
 */
class Alerts extends React.PureComponent {
    static timeout = null;

    static propTypes = {
        /** @ignore */
        disposeOffAlert: PropTypes.func.isRequired,
        /** @ignore */
        alerts: PropTypes.object.isRequired,
    };

    componentWillReceiveProps(nextProps) {
        if (this.timeout) {
            clearTimeout(this.timeout);
        }
        if (nextProps.alerts.category.length && nextProps.alerts.closeInterval > 0) {
            this.timeout = setTimeout(() => {
                this.props.disposeOffAlert();
            }, nextProps.alerts.closeInterval);
        }
    }

    closeAlert() {
        if (this.timeout) {
            clearTimeout(this.timeout);
        }
        this.props.disposeOffAlert();
    }

    render() {
        const { alerts, disposeOffAlert } = this.props;

        return (
            <div className={css.wrapper}>
                <div
                    onClick={() => disposeOffAlert()}
                    className={classNames(alerts.category.length ? css.visible : null, css[`${alerts.category}`])}
                >
                    <span>
                        <Icon icon="cross" size={14} />
                    </span>
                    {alerts.title && <h2>{alerts.title}</h2>}
                    {alerts.message && <p>{alerts.message}</p>}
                </div>
            </div>
        );
    }
}

export default withAlertsData(Alerts);
