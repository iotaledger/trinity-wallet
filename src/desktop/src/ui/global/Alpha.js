import React from 'react';

import Modal from 'ui/components/modal/Modal';
import Button from 'ui/components/Button';

/**
 * Alpha release warning modal
 */
class AlphaWarning extends React.PureComponent {
    state = {
        active: true,
    };

    render() {
        const { active } = this.state;

        return (
            <Modal variant="global" isOpen={active} onClose={() => {}}>
                <section>
                    <article>
                        <h1 style={{ color: 'var(--negative)' }}>Alpha release warning</h1>
                        <p>
                            This is an early (alpha) release of the wallet. It is to be used for testing purposes only,
                            with very small amounts of tokens.
                        </p>
                    </article>
                    <footer>
                        <Button onClick={() => this.setState({ active: false })} variant="dark">
                            Continue
                        </Button>
                    </footer>
                </section>
            </Modal>
        );
    }
}

export default AlphaWarning;
