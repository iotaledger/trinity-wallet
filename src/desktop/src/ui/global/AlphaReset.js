/*global Electron*/
import React from 'react';

import { DESKTOP_VERSION } from 'config';

import Button from 'ui/components/Button';

import css from 'ui/views/onboarding/index.scss';

/**
 * Alpha version activation view
 */
class Activation extends React.PureComponent {
    resetWallet = (e) => {
        e.preventDefault();

        localStorage.clear();

        Electron.clearStorage();
        Electron.setActiveVersion(DESKTOP_VERSION);

        location.reload();
    };

    render() {
        return (
            <main className={css.onboarding}>
                <header />
                <section>
                    <form className="center" onSubmit={this.resetWallet}>
                        <fieldset>
                            <h2>This alpha release requires a full wallet reset!</h2>
                            <p>The wallet will be reset to defaults and all your saved data will be lost.</p>
                            <Button type="submit" variant="primary">
                                Reset wallet
                            </Button>
                        </fieldset>
                    </form>
                </section>
                <footer />
            </main>
        );
    }
}

export default Activation;
