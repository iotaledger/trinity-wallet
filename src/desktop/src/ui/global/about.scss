.about {
    article {
        display: block;
        position: relative;
        max-width: 680px;
        max-height: 300px;
        width: calc(100% - 20px);
        height: calc(100vh - 200px);
        margin: 40px auto;
        text-align: left;
        background: var(--box-bg);
        padding: 20px 10px 20px 20px;
    }

    h1 {
        margin: 20px auto;
        line-height: 18px;
        font-size: 38px;
    }

    p {
        font-size: 14px;
        strong {
            color: var(--negative);
        }
    }

    h2 {
        display: inline-block;
        color: var(--primary);
        font-size: 18px;
        margin-bottom: 0px;

        small {
            margin: 5px 0 0 6px;
            float: right;
            font-size: 11px;
            padding: 0 10px 1px;
            border-radius: 3px;
            background: #c65b38;
            color: #fff;
        }
    }

    h5 {
        font-size: 18px;
        font-weight: 400;
        margin-bottom: 8px;
    }
    ul {
        margin-bottom: 20px;
    }

    li {
        font-size: 14px;
        line-height: 21px;
        margin: 0 0 4px 24px;
        list-style: circle;
    }
}
