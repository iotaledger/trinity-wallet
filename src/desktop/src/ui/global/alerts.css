.wrapper {
    position: fixed;
    right: 0;
    top: 0;
    z-index: 1002;

    > div {
        @media print {
            display: none;
        }

        position: relative;
        margin: 8px;
        padding: 8px 16px 16px 24px;
        width: 320px;
        background-color: var(--body-bg);
        border: none;
        max-width: calc(100% - 32px);

        h2 {
            margin: 0;
            margin-bottom: 4px;
        }

        span {
            cursor: pointer;
            position: absolute;
            top: 0px;
            right: 0px;
            font-size: 16px;
            line-height: 1;
            padding: 12px;
        }

        &.info {
            background: var(--positive);
            color: var(--positive-body);
        }

        &.error {
            background: var(--negative);
            color: var(--negative-body);
        }

        &.success {
            background: var(--positive);
            color: var(--positive-body);
        }

        &.extra {
            background: var(--extra);
            color: var(--extra-body);
        }

        &.warning {
            background: var(--negative);
            color: var(--negative-body);
        }
    }
}
