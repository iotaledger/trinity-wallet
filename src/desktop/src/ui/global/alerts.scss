.wrapper {
    position: fixed;
    right: 0;
    top: 0;
    z-index: 1002;
    pointer-events: none;

    > div {
        @media print {
            display: none;
        }

        display: block;
        position: relative;
        margin: 50px 20px 0;
        padding: 18px 32px 24px 24px;
        min-width: 340px;
        width: 320px;
        background: var(--body-bg);
        max-width: calc(100% - 32px);
        min-height: 120px;
        transform: scale(0.8);
        opacity: 0;
        pointer-events: none;
        border-radius: 6px;

        * {
            cursor: pointer;
        }

        h2 {
            margin: 0;
            font-size: 21px;
            margin-bottom: 16px;
            font-weight: 400;
        }

        p {
            font-size: 16px;
            line-height: 24px;
            margin-bottom: 0px;
            word-break: break-word;
        }

        span {
            cursor: pointer;
            position: absolute;
            top: 6px;
            right: 6px;
        }

        &.visible {
            opacity: 1;
            pointer-events: all;
            cursor: pointer;
            transform: scale(1);
            transition: opacity 0.12s ease-in, transform 0.24s cubic-bezier(0.23, 1, 0.32, 1);
        }

        &.info {
            background: var(--positive);
            color: var(--positive-body);
            border-color: var(--positive-hover);
            span:hover {
                color: var(--positive-hover);
            }
        }

        &.error {
            background: var(--negative);
            color: var(--negative-body);
            border-color: var(--negative-hover);
            span:hover {
                color: var(--negative-hover);
            }
        }

        &.success {
            background: var(--positive);
            color: var(--positive-body);
            border-color: var(--positive-hover);
            span:hover {
                color: var(--positive-hover);
            }
        }
    }
}
