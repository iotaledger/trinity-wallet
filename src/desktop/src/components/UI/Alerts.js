import React from 'react';
import { connect } from 'react-redux';
import { translate } from 'react-i18next';
import { disposeOffAlert } from 'actions/alerts';
import PropTypes from 'prop-types';
import Modal from 'components/UI/Modal';
import Button from 'components/UI/Button';

class Alerts extends React.PureComponent {
    static propTypes = {
        disposeOffAlert: PropTypes.func.isRequired,
        alerts: PropTypes.object.isRequired,
        t: PropTypes.func.isRequired,
    };

    static defaultProps = {
        notifications: {},
    };

    state = {};

    render() {
        const { alerts, disposeOffAlert, t } = this.props;

        if (!alerts.category.length) {
            return null;
        }

        return (
            <Modal isConfirm isOpen onClose={disposeOffAlert}>
                <h1 className={alerts.category}>{alerts.title}</h1>
                <p>{alerts.message}</p>
                <Button onClick={disposeOffAlert} variant="success">
                    {t('global:back')}
                </Button>
            </Modal>
        );
    }
}

const mapStateToProps = (state) => ({
    alerts: state.alerts,
});

const mapDispatchToProps = {
    disposeOffAlert,
};

export default translate('alerts')(connect(mapStateToProps, mapDispatchToProps)(Alerts));
