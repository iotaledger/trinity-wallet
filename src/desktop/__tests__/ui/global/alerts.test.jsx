import React from 'react';
import { shallow } from 'enzyme';

import { AlertsComponent as Alerts } from 'ui/global/Alerts';

const props = {
    dismissAlert: jest.fn(),
    alerts: {
        title: 'Foo',
        message: 'Bar',
        category: 'Baz',
    },
    forceUpdate: false,
    shouldUpdate: false,
    displayTestWarning: false,
    t: (str) => str,
};

global.Electron = {
    getOS: () => {},
    onEvent: function(_event, e) {
        this.updateEvent = e;
    },
    removeEvent: () => {},
    updateEvent: null,
};

describe('Alerts component', () => {
    test('Render the component', () => {
        const wrapper = shallow(<Alerts {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Alerts visible', () => {
        const wrapper = shallow(<Alerts {...props} />);

        expect(wrapper.find('.visible')).toHaveLength(1);
    });

    test('Alerts hidden', () => {
        const mockProps = Object.assign({}, props, {
            alerts: {
                title: '',
                message: '',
                category: '',
            },
        });
        const wrapper = shallow(<Alerts {...mockProps} />);

        expect(wrapper.find('.visible')).toHaveLength(0);
    });

    test('Display mandatory update', () => {
        const mockProps = Object.assign({}, props, { forceUpdate: true });
        const wrapper = shallow(<Alerts {...mockProps} />);

        expect(wrapper.find('.update')).toHaveLength(1);
        expect(wrapper.find('strong').text()).toEqual('global:forceUpdate');
    });

    test('Display optional update', () => {
        const mockProps = Object.assign({}, props, { shouldUpdate: true });
        const wrapper = shallow(<Alerts {...mockProps} />);

        expect(wrapper.find('.update')).toHaveLength(1);
        expect(wrapper.find('strong').text()).toEqual('global:shouldUpdate');
    });

    test('Hide update banner if update in progress', () => {
        const mockProps = Object.assign({}, props, { forceUpdate: true, shouldUpdate: true });
        const wrapper = shallow(<Alerts {...mockProps} />);

        Electron.updateEvent({ percent: 99 });

        expect(wrapper.find('.update')).toHaveLength(0);
    });
});
