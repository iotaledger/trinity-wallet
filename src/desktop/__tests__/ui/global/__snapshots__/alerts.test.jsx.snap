// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alerts component Render the component 1`] = `
<div
  className="wrapper"
>
  <div
    className="visible Baz"
    onClick={[Function]}
  >
    <span>
      <Icon
        icon="cross"
        size={14}
      />
    </span>
    <h2>
      Foo
    </h2>
    <p>
      Bar
    </p>
  </div>
</div>
`;
